include/irr/core/math/aabb2dSIMDIntBase.h

```diff
@@ -216,9 +175,80 @@ class aabbSIMD<core::vector2df_SIMD> : public aabbSIMDBase<aabbSIMD, core::vecto
 
 };
 
+template<typename VECTOR_TYPE>
+class aabb2dSIMDInt : public aabbSIMDBase<aabb2dSIMDInt, VECTOR_TYPE>
+{
+	static_assert(
+		std::is_same<VECTOR_TYPE, core::vector2di32_SIMD>::value ||
+		std::is_same<VECTOR_TYPE, core::vector2du32_SIMD>::value,
+		"assertion failed: cannot use this type");
+
+public:
+	inline aabb2dSIMDInt()
+	{
+		if constexpr (std::is_same<VECTOR_TYPE, core::vector2di32_SIMD>::value)
+			internalBoxRepresentation = core::vector4di32_SIMD(INT_MIN);
+
+		if constexpr (std::is_same<VECTOR_TYPE, core::vector2du32_SIMD>::value)
+			internalBoxRepresentation = core::vector4du32_SIMD(0u);
```

Comment:
(UINT_MAX,UINT_MAX,0u,0u)

Reply:
@devshgraphicsprogramming we are allowed to use `if constexpr` now?

Reply:
no, see the other comment I left about `static_if`